### PR TITLE
randomness #4: RandManager update from randomnet

### PR DESCRIPTION
**Pull request by zjma**

### CODE REVIEW

File: consensus/src/rand/rand_gen/rand_manager.rs [hunk 1]
```diff
@@ -236,11 +263,7 @@ impl<S: Share, D: AugmentedData, Storage: AugDataStorage<D>> RandManager<S, D, S
         DropGuard::new(abort_handle)
     }
 
-    async fn broadcast_aug_data(&mut self) -> CertifiedAugData<D> {
-        if let Some(certified_data) = self.aug_data_store.get_my_certified_aug_data() {
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/rand_manager.rs:240]: Why is this removed? cc @zekun000

File: consensus/src/rand/rand_gen/rand_manager.rs [hunk 2]
```diff
@@ -364,4 +410,16 @@ impl<S: Share, D: AugmentedData, Storage: AugDataStorage<D>> RandManager<S, D, S
         }
         info!("RandManager stopped");
     }
+
+    pub fn observe_queue(&self) {
+        let queue = &self.block_queue.queue;
+        RAND_QUEUE_SIZE.set(queue.len() as i64);
+        info!(
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/rand_manager.rs:417]: Downgrade to debug? Or maybe we only print if the queue is too large.

File: consensus/src/rand/rand_gen/reliable_broadcast_state.rs [hunk 3]
```diff
@@ -34,18 +38,23 @@ impl<D> AugDataCertBuilder<D> {
...
-impl<S: Share, D: AugmentedData> BroadcastStatus<RandMessage<S, D>, RandMessage<S, D>>
+impl<S: TShare, D: TAugmentedData> BroadcastStatus<RandMessage<S, D>, RandMessage<S, D>>
     for Arc<AugDataCertBuilder<D>>
 {
     type Aggregated = CertifiedAugData<D>;
     type Message = AugData<D>;
     type Response = AugDataSignature;
 
     fn add(&self, peer: Author, ack: Self::Response) -> anyhow::Result<Option<Self::Aggregated>> {
+        debug!(
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/reliable_broadcast_state.rs:49]: We may want to clean up these logs later..

**Review by danielxiangzl — APPROVED** [review]

File: consensus/src/rand/rand_gen/rand_manager.rs (hunk 1 above)
Comment (zjma) [reply consensus/src/rand/rand_gen/rand_manager.rs:240]: it's just moved to L279?

File: consensus/src/rand/rand_gen/block_queue.rs [hunk 4]
```diff
@@ -85,9 +86,20 @@ impl QueueItem {
 
 /// Maintain ordered blocks that have pending randomness
 pub struct BlockQueue {
-    queue: BTreeMap<Round, QueueItem>,
+    pub queue: BTreeMap<Round, QueueItem>,
```
Comment (zekun000) [inline consensus/src/rand/rand_gen/block_queue.rs:89]: why is this pub?

File: consensus/src/rand/rand_gen/rand_manager.rs [hunk 5]
```diff
@@ -297,22 +326,21 @@ impl<S: Share, D: AugmentedData, Storage: AugDataStorage<D>> RandManager<S, D, S
             )
         );
 
-        let certified_data = self.broadcast_aug_data().await;
-        let _guard = self.broadcast_certified_aug_data(certified_data);
-
+        let _guard = self.broadcast_aug_data().await;
+        let mut interval = tokio::time::interval(Duration::from_millis(5000));
```
Comment (zekun000) [inline consensus/src/rand/rand_gen/rand_manager.rs:330]: nit: just use from_secs

**Review by zekun000 — APPROVED** [review]

File: consensus/src/rand/rand_gen/reliable_broadcast_state.rs (hunk 3 above)
Comment (zjma) [reply consensus/src/rand/rand_gen/reliable_broadcast_state.rs]: fixed

File: consensus/src/rand/rand_gen/block_queue.rs (hunk 4 above)
Comment (zjma) [reply consensus/src/rand/rand_gen/block_queue.rs]: fixed

File: consensus/src/rand/rand_gen/reliable_broadcast_state.rs (hunk 3 above)
Comment (zjma) [reply consensus/src/rand/rand_gen/reliable_broadcast_state.rs]: fixed

File: consensus/src/rand/rand_gen/rand_manager.rs (hunk 2 above)
Comment (zjma) [reply consensus/src/rand/rand_gen/rand_manager.rs]: removed.
